drivers/advanced_io/advanced_io.c

```diff
@@ -8,25 +8,13 @@
 
 #include "periph/gpio.h"
 
-typedef enum {
-    MSBFIRST,
-    LSBFIRST
-} advanced_io_bit_order_t;
-
-uint8_t advanced_io_shift_in(gpio_t data_pin, gpio_t clock_pin, advanced_io_bit_order_t bit_order){
+uint8_t advanced_io_shift_in(gpio_t data_pin, gpio_t clock_pin){
     uint8_t byte = 0x00;
 
-    for(int i = 0; i <= 8; i++){
+    for(unsigned int i = 0; i <= 8; i++){
         gpio_set(clock_pin);
         if(gpio_read(data_pin) > 0){
-            if(bit_order == MSBFIRST){
-                byte = byte | 0x01 << (8 - i);
-            }
-            else if(bit_order == LSBFIRST){
-                byte = byte | 0x01 << i;
-            } else {
-                return '\0';
-            }
+            byte = byte | 0x01 << (8 - i);
```

Comment:
first iteration shifts 1 eight times to the left, which results in 0, right?

Also, if you change the for loop to count from 7 (or 8) to zero, you can simplify this line to ```byte |= (1<<i);```.

Reply:
It should be 7 times, not 8, right. I am now surprised that I get reasonable results with my sensor. True, that would be way simpler.